### PR TITLE
Support reading uv and uv map for ply format if texture_uv exists in ply file

### PR DESCRIPTION
**Pull request by YangHai-1218**
When the ply format looks as follows:
  ```
comment TextureFile ***.png
element vertex 892
property double x
property double y
property double z
property double nx
property double ny
property double nz
property double texture_u
property double texture_v
```
`MeshPlyFormat` class will read uv from the ply file and read the uv map as commented as TextureFile.

### CODE REVIEW

File: pytorch3d/renderer/mesh/rasterizer.py [hunk 1]
```diff
@@ -110,7 +110,9 @@ def __init__(self, cameras=None, raster_settings=None) -> None:
 
     def to(self, device):
         # Manually move to device cameras as it is not a subclass of nn.Module
-        self.cameras = self.cameras.to(device)
+        cameras = self.cameras
+        if cameras is not None:
```
Comment (bottler) [inline pytorch3d/renderer/mesh/rasterizer.py:114]: This looks like a significant bug you are fixing, which is separate from this change. Thank you. It would be good to put it in its own PR which would include a test for this problem. 

(Or let us know and we can do it internally.)

File: pytorch3d/renderer/mesh/textures.py [hunk 2]
```diff
@@ -1073,7 +1073,7 @@ def join_batch(self, textures: List["TexturesUV"]) -> "TexturesUV":
         faces_uvs_list += self.faces_uvs_list()
         verts_uvs_list += self.verts_uvs_list()
         maps_list += self.maps_list()
-        num_faces_per_mesh = self._num_faces_per_mesh
+        num_faces_per_mesh = self._num_faces_per_mesh.copy()
```
[the remark this answers is not in the document]
Comment (bottler) [reply pytorch3d/renderer/mesh/textures.py:1076]: In fact, I think TexturesAtlas has exactly the same bug. They should be fixed at the same time.

[the remark this answers is not in the document]
Comment (YangHai-1218) [reply pytorch3d/renderer/mesh/textures.py:1076]: Yes, this is a significant bug which was found when I was using Pytorch3D. Specifically, for  meshes with TextureUV, calling function `join_meshes_as_batch` will cause the origin mesh textureUV's `_num_faces_per_mesh` changed, due to the list reference attribute. Especially when calling `join_meshes_per_mesh` multiple times, this will cause OOM finally, because the original mesh textureUV's `_num_faces_per_mesh` list has became too long. I have not noticed the TextureAltas has the same bug because I have not used it. This is just a small change, so I committed it together. Can you please sort it out if needed?

**Comment by YangHai-1218** [comment]
> Thanks. This looks like the right logic.
> 
> Do you have an example of data in this format which we can use as a test for this new functionality? (It would need to be liberally licensed.) Otherwise, can you create a small synthetic example?
> 
> In particular: The UV coordinates are sent directly from the PLY to TexturesUV without changing their range or direction. Without a real example, I cannot tell whether that is correct.

This is a test mesh, which contains a ply file and its texture file. It is licensed. You can test the code with it.
[sample_mesh.zip](https://github.com/facebookresearch/pytorch3d/files/8182684/sample_mesh.zip)

File: pytorch3d/renderer/mesh/textures.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bottler) [reply pytorch3d/renderer/mesh/textures.py:1076]: Yes. Let me sort it out. Best to remove this change from this PR still.

**Comment by bottler** [comment]
What do you mean by "it is licensed"? We need to use an existing example with a known permissive license, or to create a new example. (It can then be added to tests/data and a test validating what it looks like against an image can be created. This should happen in this PR.)

**Comment by YangHai-1218** [comment]
Its license is MIT. Is this enough? If not, I will create an example to do testing.

**Comment by bottler** [comment]
Thanks!
